### ISSUE TITLE
runner_af2advanced.py gets stuck?

### ISSUE THREAD

**Issue by A-Talavera** [YoshitakaMo/localcolabfold#20]
Dear all,

I am running the following command:

export NVIDIA_VISIBLE_DEVICES='all'
export TF_FORCE_UNIFIED_MEMORY='1'
export XLA_PYTHON_CLIENT_MEM_FRACTION='4.0'
colabfold-conda/bin/python3.7 runner_af2advanced.py --input RctB.fasta   --output_dir RctB_monomer --max_recycle 3 --use_ptm --use_turbo --num_relax Top1

The sequence has 658 residues and it has been running for more than 12 hours on a 4xGPUs machine (RTX 2080Ti with 12 GB). 

ColabFold on Linux
2021-10-09 13:03:38.900883: I tensorflow/stream_executor/platform/default/dso_loader.cc:53] Successfully opened dynamic library libcudart.so.11.0
2021-10-09 13:03:39.713378: I tensorflow/stream_executor/platform/default/dso_loader.cc:53] Successfully opened dynamic library libcuda.so.1
2021-10-09 13:03:39.734361: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1733] Found device 0 with properties: 
pciBusID: 0000:19:00.0 name: NVIDIA GeForce RTX 2080 Ti computeCapability: 7.5
coreClock: 1.545GHz coreCount: 68 deviceMemorySize: 10.76GiB deviceMemoryBandwidth: 573.69GiB/s
2021-10-09 13:03:39.735057: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1733] Found device 1 with properties: 
pciBusID: 0000:1a:00.0 name: NVIDIA GeForce RTX 2080 Ti computeCapability: 7.5
coreClock: 1.545GHz coreCount: 68 deviceMemorySize: 10.76GiB deviceMemoryBandwidth: 573.69GiB/s
2021-10-09 13:03:39.735707: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1733] Found device 2 with properties: 
pciBusID: 0000:67:00.0 name: NVIDIA GeForce RTX 2080 Ti computeCapability: 7.5
coreClock: 1.545GHz coreCount: 68 deviceMemorySize: 10.76GiB deviceMemoryBandwidth: 573.69GiB/s
2021-10-09 13:03:39.736356: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1733] Found device 3 with properties: 
pciBusID: 0000:68:00.0 name: NVIDIA GeForce RTX 2080 Ti computeCapability: 7.5
coreClock: 1.545GHz coreCount: 68 deviceMemorySize: 10.76GiB deviceMemoryBandwidth: 573.69GiB/s
2021-10-09 13:03:39.736390: I tensorflow/stream_executor/platform/default/dso_loader.cc:53] Successfully opened dynamic library libcudart.so.11.0
2021-10-09 13:03:39.738905: I tensorflow/stream_executor/platform/default/dso_loader.cc:53] Successfully opened dynamic library libcublas.so.11
2021-10-09 13:03:39.738953: I tensorflow/stream_executor/platform/default/dso_loader.cc:53] Successfully opened dynamic library libcublasLt.so.11
2021-10-09 13:03:39.739855: I tensorflow/stream_executor/platform/default/dso_loader.cc:53] Successfully opened dynamic library libcufft.so.10
2021-10-09 13:03:39.740062: I tensorflow/stream_executor/platform/default/dso_loader.cc:53] Successfully opened dynamic library libcurand.so.10
2021-10-09 13:03:39.742704: I tensorflow/stream_executor/platform/default/dso_loader.cc:53] Successfully opened dynamic library libcusolver.so.11
2021-10-09 13:03:39.743337: I tensorflow/stream_executor/platform/default/dso_loader.cc:53] Successfully opened dynamic library libcusparse.so.11
2021-10-09 13:03:39.743445: I tensorflow/stream_executor/platform/default/dso_loader.cc:53] Successfully opened dynamic library libcudnn.so.8
2021-10-09 13:03:39.749167: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1871] Adding visible gpu devices: 0, 1, 2, 3
Input ID: RctB_V.Cholereae
Input Sequence: 
homooligomer: 1
total_length: 658
output_dir: RctB_monomer
running mmseqs2
  0%|          | 0/5 [elapsed: 00:00 remaining: ?]
2021-10-09 13:03:44.719056: I tensorflow/core/platform/cpu_feature_guard.cc:142] This TensorFlow binary is optimized with oneAPI Deep Neural Network Library (oneDNN) to use the following CPU instructions in performance-critical operations:  AVX2 AVX512F FMA
To enable them in other operations, rebuild TensorFlow with the appropriate compiler flags.
2021-10-09 13:03:44.721174: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1258] Device interconnect StreamExecutor with strength 1 edge matrix:
2021-10-09 13:03:44.721200: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1264]      
2021-10-09 13:03:44.762135: I tensorflow/core/platform/profile_utils/cpu_utils.cc:114] CPU Frequency: 3100000000 Hz
model_1_ptm_seed_0 recycles:3 tol:6.20 pLDDT:54.26 pTMscore:0.30

I get figures 1 and 2. From figure 2 it seems that it already got a decent model but I do not have any pdb file. Moreover, after a command "top" I cannot see any process related to the run. and after "nvidia-smi" I get this:

![Screenshot from 2021-10-09 14-39-35](https://user-images.githubusercontent.com/85237041/136658254-bd3b63a9-0e6a-4bcc-8217-6470f97a84b0.png)

I can see that colabfold-conda/bin.python3.7 is running in all four GPUS, but only the memory of GPU 0 is currently been relatively used, but at the same time the GPU-Utilization is 0 %. Is this the normal behaviour or I am doing something wrong?

Thanks a lot in advance.
Best regards
Ariel

**Comment by YoshitakaMo** [comment]
I have no idea yet why the pdb file is not being output.
But if you know that the model1 seems to be good prediction, then you probably don't need to compute 5 models. Currently the random seed for each model used in colabfold/localcolabfold is fixed, so it will return the same results on the same machine. Therefore, if you set `--num_models=1`, only the first model will be calculated.

However,
> model_1_ptm_seed_0 recycles:3 tol:6.20 pLDDT:54.26 pTMscore:0.30

The displayed tol, pLDDT and pTMscore values suggest that the predicted model is not good yet. The values may be improved if you increase the value of `--max_recycles`.
